### PR TITLE
feat: templates video preview

### PR DESCRIPTION
**Pull request by Kneesal**
# Description

### <samp>🤖 Generated by Copilot at 941ca49</samp>

This pull request adds a new feature to the journey template details page, which allows users to see different previews of a journey template with cards and videos. It introduces a new `TemplatePreviewTabs` component, which uses the `Swiper` library and the `TabPanel` component to render a tabbed interface for the previews. It also refactors the template details page to use a new `PageWrapper` component and a `Stack` component for the layout, and to display the template title in the page header and SEO metadata. It includes tests, stories, and data files for the new component.

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/34356579/todos/6557560976)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] it should render video preview images

# Walkthrough

### <samp>🤖 Generated by Copilot at 941ca49</samp>

*  Add the TemplatePreviewTabs component to render a tabbed interface for previewing the cards and videos of a journey template ([link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-501634969741d25ebfcafc40c5ccc588acb4eaaeb397d9b5d56f541f7a38c08fR1-R105),[link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-080d8aa134f161e27b912696140af965864ce69584482aa845bfcec696bfa65aR1-R61),[link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-15fb153f0861beb6dce8cb73e1de5166f1310bbce3e3a0f3d6e4960374c96bbfR1-R76),[link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-260dd2a89b552ea7ac4bae5a45ea16adf3ef09bb2e8bcc78aab183da8a2e3f7aR1),[link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-e41ee5182dac228c799695276d845c2f69735e29a2ae299c70b6c07ec81ec512R1-R289))
  * Use the Swiper library for the video carousel and the TabPanel component for the tab content in `TemplatePreviewTabs.tsx` ([link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-501634969741d25ebfcafc40c5ccc588acb4eaaeb397d9b5d56f541f7a38c08fR1-R105))
  * Showcase the component with different journeys and allow interactive testing and documentation in `TemplatePreviewTabs.stories.tsx` ([link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-080d8aa134f161e27b912696140af965864ce69584482aa845bfcec696bfa65aR1-R61))
  * Render the component with different journeys and check the functionality and accessibility of the tabs and the previews in `TemplatePreviewTabs.spec.tsx` ([link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-15fb153f0861beb6dce8cb73e1de5166f1310bbce3e3a0f3d6e4960374c96bbfR1-R76))
  * Export the component for easy importing from other files in `index.ts` ([link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-260dd2a89b552ea7ac4bae5a45ea16adf3ef09bb2e8bcc78aab183da8a2e3f7aR1))
  * Add a mock journey with video blocks for testing and story purposes in `data.ts` ([link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-e41ee5182dac228c799695276d845c2f69735e29a2ae299c70b6c07ec81ec512R1-R289))
* Update the TemplateView component to display the preview tabs below the template title and menu ([link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-436738bb5e18937497677275cf375e4dbebb3183971ed6225561bed0c82bd150R7-R8),[link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-436738bb5e18937497677275cf375e4dbebb3183971ed6225561bed0c82bd150R16))
  * Import the TemplatePreviewTabs component from its index file in `TemplateView.tsx` ([link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-436738bb5e18937497677275cf375e4dbebb3183971ed6225561bed0c82bd150R7-R8))
  * Pass the journey data as a prop to the TemplatePreviewTabs component in `TemplateView.tsx` ([link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-436738bb5e18937497677275cf375e4dbebb3183971ed6225561bed0c82bd150R16))
* Refactor the PageWrapper component to support custom children for the main header and update the template details page to use the new version ([link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-46a8849cb7583f95a436ff12b776363786b82cc424ad3415052b8c7cd85052ccL16-R17),[link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-46a8849cb7583f95a436ff12b776363786b82cc424ad3415052b8c7cd85052ccL46-R59))
  * Update the import of the PageWrapper component from the NewPageWrapper file in `[journeyId].tsx` ([link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-46a8849cb7583f95a436ff12b776363786b82cc424ad3415052b8c7cd85052ccL16-R17))
  * Pass the Menu component as a child of a Stack component to the mainHeaderChildren prop instead of the menu prop in `[journeyId].tsx` ([link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-46a8849cb7583f95a436ff12b776363786b82cc424ad3415052b8c7cd85052ccL46-R59))
* Update the title of the NextSeo and PageWrapper components to use the template title or a fallback value of 'Template Details' instead of 'Journey Template' in `[journeyId].tsx` ([link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-46a8849cb7583f95a436ff12b776363786b82cc424ad3415052b8c7cd85052ccL36-R37),[link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-46a8849cb7583f95a436ff12b776363786b82cc424ad3415052b8c7cd85052ccL46-R59))
* Import the Stack component from the MUI library to create a horizontal layout for the menu in the template details page in `[journeyId].tsx` ([link](https://github.com/JesusFilm/core/pull/1937/files?diff=unified&w=0#diff-46a8849cb7583f95a436ff12b776363786b82cc424ad3415052b8c7cd85052ccR1))

### CODE REVIEW

File: apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplatePreviewTabs.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      <TabPanel name="videos-preview-tab" value={tabValue} index={1}>
+        <Swiper
+          grabCursor
+          freeMode
+          slidesPerView="auto"
+          spaceBetween={24}
+          slidesOffsetAfter={64}
+          style={{
+            paddingTop: '20px',
+            // width: '130%',
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplatePreviewTabs.tsx:83]: can remove?

File: apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplatePreviewTabs.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import {
+  GetJourney_journey_blocks_StepBlock as StepBlock,
+  GetJourney_journey_blocks_VideoBlock as VideoBlock
+} from '../../../../__generated__/GetJourney'
+import { CardPreview } from '../../CardPreview'
+
+export function TemplatePreviewTabs(): ReactElement {
+  const [tabValue, setTabValue] = useState(0)
+  const { t } = useTranslation('apps-journeys-admin')
+  const smUp = useMediaQuery((theme: Theme) => theme.breakpoints.up('sm'))
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplatePreviewTabs.tsx:28]: Is it possible for you to try to do same logic you've applied below without using `useMediaQuery`. UseMediaQuery from memory affects performance quiet a bit

File: apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplatePreviewTabs.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        >
+          {videoBlocks?.map((block) => (
+            <SwiperSlide key={block.id} style={{ width: 423 }}>
+              <NextImage
+                width={423}
+                height={239}
+                key={block.id}
+                sx={{ borderRadius: 4, objectFit: 'cover' }}
+                src={(block.image as string) ?? block.video?.image}
+              />
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplatePreviewTabs.tsx:96]: Correct me if I'm wrong. But I thought that from the conversation this morning we are to render a video?

File: apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplatePreviewTabs.tsx [hunk 4]
```diff
@@ -0,0 +1,114 @@
+import 'swiper/swiper.min.css'
+import 'swiper/components/scrollbar/scrollbar.min.css'
```
Comment (Kneesal) [inline apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplatePreviewTabs.tsx:2]: a reminder to remove this

**Pull request closed without merging**